tests/enterprise-contract/contract.go

```diff
@@ -0,0 +1,142 @@
+	publicSecretName := "cosign-public-key"
+
+	BeforeAll(func() {
+		fwk, err = framework.NewFramework(constants.TEKTON_CHAINS_E2E_USER)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fwk.UserNamespace).NotTo(BeNil(), "failed to create sandbox user")
+		namespace = fwk.UserNamespace
+		kubeController = tekton.KubeController{
+			Commonctrl: *fwk.AsKubeAdmin.CommonController,
+			Tektonctrl: *fwk.AsKubeAdmin.TektonController,
+			Namespace:  namespace,
+		}
+		publicKey, err := kubeController.GetTektonChainsPublicKey()
+		Expect(err).ToNot(HaveOccurred())
+		GinkgoWriter.Printf("Copy public key from %s/signing-secrets to a new secret\n", constants.TEKTON_CHAINS_NS)
+		Expect(kubeController.CreateOrUpdateSigningSecret(
+			publicKey, publicSecretName, namespace)).To(Succeed())
+
+		defaultEcp, err := kubeController.GetEnterpriseContractPolicy("default", "enterprise-contract-service")
+		Expect(err).NotTo(HaveOccurred())
+		policySource = defaultEcp.Spec.Sources
+
+		cm, err := kubeController.Commonctrl.GetConfigMap("ec-defaults", "enterprise-contract-service")
+		Expect(err).ToNot(HaveOccurred())
+		verifyECTaskBundle = cm.Data["verify_ec_task_bundle"]
+		Expect(verifyECTaskBundle).ToNot(BeEmpty())
+		GinkgoWriter.Printf("Using verify EC task bundle: %s\n", verifyECTaskBundle)
+
+	})
+
+	Context("Release Policy", func() {
+		BeforeAll(func() {
+			generator = tekton.VerifyEnterpriseContract{
+				Bundle:              verifyECTaskBundle,
+				Image:               imageWithDigest,
+				Name:                "verify-enterprise-contract",
+				Namespace:           namespace,
+				PolicyConfiguration: "ec-policy",
+				PublicKey:           fmt.Sprintf("k8s://%s/%s", namespace, publicSecretName),
+				SSLCertDir:          "/var/run/secrets/kubernetes.io/serviceaccount",
+				Strict:              true,
+			}
+			secretName := fmt.Sprintf("golden-image-public-key%s", util.GenerateRandomString(10))
+			publicKey := []byte("-----BEGIN PUBLIC KEY-----\n" +
+				"MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEZP/0htjhVt2y0ohjgtIIgICOtQtA\n" +
+				"naYJRuLprwIv6FDhZ5yFjYUEtsmoNcW7rx2KM6FOXGsCX3BNc7qhHELT+g==\n" +
+				"-----END PUBLIC KEY-----")
+
+			GinkgoWriter.Println("Create golden image public signing key")
+			Expect(kubeController.CreateOrUpdateSigningSecret(publicKey, secretName, namespace)).To(Succeed())
+			generator.PublicKey = fmt.Sprintf("k8s://%s/%s", namespace, secretName)
+		})
+
+		It("Release policy: Task bundle is not acceptable", func() {
+			policy := ecp.EnterpriseContractPolicySpec{
+				Sources: policySource,
+				Configuration: &ecp.EnterpriseContractPolicyConfiguration{
+					Include: []string{"attestation_task_bundle.unacceptable_task_bundle"},
+				},
+			}
+			Expect(kubeController.CreateOrUpdatePolicyConfiguration(namespace, policy)).To(Succeed())
+
+			generator.Image = "quay.io/redhat-appstudio/ec-golden-image:08b938499ee47a25673f5e6a56f69ce57d87b45c"
+			pr, err := kubeController.RunPipeline(generator, pipelineRunTimeout)
```

Comment:
I saw that variable is declared in line 24, but I don't see where you set the value of the time value

Reply:
thanks @kasemAlem , added the definition `pipelineRunTimeout = int(time.Duration(5) * time.Minute)`